packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/analytics_client.dart

```diff
@@ -0,0 +1,92 @@
+import 'dart:async';
+
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/session_manager.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/shared_prefs.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/stoppable_timer.dart';
+import 'package:amplify_core/amplify_core.dart';
+
+import '../../sdk/pinpoint.dart';
+
+import 'endpoint_client/endpoint_client.dart';
+import 'event_client/event_client.dart';
+import 'event_creator/event_creator.dart';
+
+class AnalyticsClient {
+  // Should be managed by SessionTracker
+  late final SessionManager _sessionManager; // = SessionManager();
+
+  late final EventClient _eventManager;
+
+  late final StoppableTimer _autoEventSubmitter;
+
+  late final EventCreator _eventClient;
```

Comment:
The names here are hard to follow. `_eventManager` is of type `EventClient`, and `_eventClient` is of type `EventCreator`. I would suggest renaming these.